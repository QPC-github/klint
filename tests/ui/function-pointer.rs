#![crate_type="lib"]

#[klint::preempt_count(adjust = 1, unchecked)]
fn spin_lock() {}

fn okay() {

}

fn not_okay() {
    spin_lock();
}

#[klint::preempt_count(adjust = 0)]
pub fn good() {
    let a: fn() = okay;
    a();
}

#[klint::preempt_count(adjust = 0)]
pub fn bad() {
    let a: fn() = not_okay;
    a();
}
