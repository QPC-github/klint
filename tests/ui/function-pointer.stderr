warning: converting this function to pointer may result in preemption count rule violation
  --> $DIR/function-pointer.rs:22:19
   |
22 |     let a: fn() = not_okay;
   |                   ^^^^^^^^
   |
   = help: `not_okay` is being converted to a pointer
   = help: adjustment of this function is inferred to be 1 and expectation is inferred to be 0..
   = help: while the adjustment for function pointers is assumed to be 0 and the expectation be 0

warning: 1 warning emitted

